src/parser/shadowlands/modules/items/dungeons/InscrutableQuantumDevice.tsx

```diff
@@ -0,0 +1,311 @@
+  protected statTracker!: StatTracker;
+
+  private readonly item: Item;
+
+  private readonly counts = {
+    crit: 0,
+    haste: 0,
+    mastery: 0,
+    vers: 0,
+
+    heal: 0,
+    execute: 0,
+
+    mana: 0,
+    ccBreak: 0,
+    get decoy() {
+      return this.decoyIds.size;
+    },
+    decoyIds: new Set<string>(),
+  };
+
+  constructor(
+    options: Options & {
+      abilities: Abilities;
+      buffs: Buffs;
+      castEfficiency: CastEfficiency;
+      statTracker: StatTracker;
+    },
+  ) {
+    super(options);
+
+    this.item = this.selectedCombatant.getItem(ITEMS.INSCRUTABLE_QUANTUM_DEVICE.id)!;
+    if (this.item == null) {
+      this.active = false;
+      return;
+    }
+
+    // https://wowhead.com/spell=330366/inscrutable-quantum-device?ilvl=262
+    const secondaryStat = calculateSecondaryStatDefault(262, 850, this.item.itemLevel);
+
+    // Add all the buffs to the statTracker so that other modules know them
+    options.statTracker.add(SPELLS.INSCRUTABLE_QUANTUM_DEVICE_CRIT, {
+      crit: secondaryStat,
+    });
+    options.statTracker.add(SPELLS.INSCRUTABLE_QUANTUM_DEVICE_HASTE, {
+      haste: secondaryStat,
+    });
+    options.statTracker.add(SPELLS.INSCRUTABLE_QUANTUM_DEVICE_MASTERY, {
+      mastery: secondaryStat,
+    });
+    options.statTracker.add(SPELLS.INSCRUTABLE_QUANTUM_DEVICE_VERS, {
+      versatility: secondaryStat,
+    });
+
+    // Add the buffs to the buff tracker so that they show up in the timeline
+    options.buffs.add({
+      spellId: IQD_BUFFS.map(({ id }) => id),
+      timelineHighlight: true,
+      triggeredBySpellId: IQD_CAST.id,
+    });
+
+    // Add the main cast as an ability so that we can track cooldown and usage in timeline
+    options.abilities.add({
+      spell: IQD_CAST.id,
+      category: Abilities.SPELL_CATEGORIES.ITEMS,
+      cooldown: COOLDOWN_SECONDS,
+      gcd: null,
+      castEfficiency: {
+        suggestion: true,
+        recommendedEfficiency: 0.8,
+      },
+    });
+
+    this.addEventListener(Events.applybuff.by(SELECTED_PLAYER).spell(IQD_BUFFS), this.onBuff);
+    this.addEventListener(
+      Events.resourcechange.by(SELECTED_PLAYER).spell(IQD_MANA),
+      () => (this.counts.mana += 1),
+    );
+    this.addEventListener(Events.cast.by(SELECTED_PLAYER_PET).spell(IQD_DECOY_DISTRACT), (e) =>
+      this.counts.decoyIds.add(encodeTargetString(e.sourceID, e.sourceInstance)),
+    );
+    this.addEventListener(
+      Events.damage.by(SELECTED_PLAYER).spell(IQD_EXECUTE),
+      () => (this.counts.execute += 1),
+    );
+    this.addEventListener(
+      Events.heal.by(SELECTED_PLAYER).spell(IQD_HEAL),
+      () => (this.counts.heal += 1),
+    );
+    this.addEventListener(Events.fightend, () => {
+      const { casts = -1 } = this.getCastEfficiency() || {};
+      const { crit, haste, mastery, vers, heal, execute, mana, ccBreak, decoy } = this.counts;
+      const total = crit + haste + mastery + vers + heal + execute + mana + ccBreak + decoy;
```

Comment:
was going to suggest using `Object.values(this.counts).reduce((a, b) => a + b)` but `decoyIds` is part of the `counts` map :thinking: 

since this is a trinket that won't be around for that much longer I think we don't worry about that.

Reply:
yeah I liked the idea of having `decoy` as a get so that you get all values in the same way, but then I wasn't allowed to access a `Set` on the `class`. I'm sorry ;P